src/App.vue

```diff
@@ -128,33 +128,53 @@ export default {
 	},
 
 	methods: {
-		loadNotes() {
-			fetchNotes()
-				.then(data => {
-					if (data === null) {
-						// nothing changed
-						return
-					}
-					if (data.notes !== null) {
-						this.error = false
-						this.routeDefault(data.lastViewedNote)
-					} else if (this.loading.notes) {
-						// only show error state if not loading in background
-						this.error = data.errorMessage
-					} else {
-						console.error('Server error while updating list of notes: ' + data.errorMessage)
-					}
-				})
-				.catch(() => {
+		async loadNotes() {
+			console.log('[App.loadNotes] Starting initial load')
+			// Skip refresh if in search mode - search results should not be overwritten
+			const searchText = store.state.app.searchText
+			if (searchText && searchText.trim() !== '') {
+				console.log('[App.loadNotes] Skipping - in search mode with query:', searchText)
```

Comment:
When the search text is cleared, we should see the list of all notes. Currently, when that happens, it's empty